### PR TITLE
Add initial support for Vector64 on Arm64 and Amd64

### PR DESCRIPTION
**Pull request by fanyang-mono**
Contributes to #42350

This change:

- Conditionally marks Vector64<T> as a SIMD type. Vector64<T> is treated as a
SIMD type on amd64 and arm64 only when the LLVM backend is in use. Vector128<T>
continues to be supported with both LLVM and mini on amd64, as before; on arm64,
it continues to only be supported with LLVM.

- Checks vector widths for compatibility when bitcasting.

- Slightly cleans up return value handling for SIMD value types--conversion
from an aggregate to an LLVM vector-typed value now happens in one location.

- Removes a 128-bit SIMD assumption in the lowering code for OP_SETRET.

- Renames `emit_vector128` to `emit_sri_vector` and makes it width-agnostic.

- Improves the generated IR for non-broadcasting Vector64/Vector128.Create; the
resulting code works entirely in registers and does not depend on temporary
stack storage for writing values into individual vector lanes.

### CODE REVIEW

File: src/mono/netcore/sample/HelloWorld/Program.cs [hunk 1]
```diff
@@ -16,4 +16,4 @@ private static void Main(string[] args)
             Console.WriteLine(System.Runtime.InteropServices.RuntimeInformation.FrameworkDescription);
         }
     }
-}
+}
```
Comment (vargaz) [inline src/mono/netcore/sample/HelloWorld/Program.cs:19]: Unrelated.

File: src/mono/mono/mini/simd-intrinsics-netcore.c [hunk 2]
```diff
@@ -261,12 +261,39 @@ get_vector_t_elem_type (MonoType *vector_type)
...
 		!strcmp (m_class_get_name (klass), "Vector256`1"));
 	etype = mono_class_get_context (klass)->class_inst->type_argv [0];
 	return etype;
 }
 
+static MonoInst *
+emit_arch_vector64_create_multi (MonoCompile *cfg, MonoMethodSignature *fsig, MonoClass *klass, MonoType *etype, MonoInst **args)
+{
+#if defined(TARGET_AMD64)
+	MonoInst *ins, *load;
```
Comment (vargaz) [inline src/mono/mono/mini/simd-intrinsics-netcore.c:275]: The commit message says Arm64.

Comment (vargaz) [reply src/mono/mono/mini/simd-intrinsics-netcore.c:275]: Would be nice to add a helper function to unify this with the emit_arch_vector128_create_multi () function, they are mostly the same except the 16/8.

**Comment by SamMonoRT** [comment]
LGTM

File: src/mono/mono/mini/mini.h [hunk 3]
```diff
@@ -316,11 +316,13 @@ enum {
...
-#define MONO_CLASS_IS_SIMD(cfg, klass) ( ((cfg)->opt & MONO_OPT_SIMD) && ( COMPILE_LLVM (cfg) ) && m_class_is_simd_type (klass) )
+/*
+ * SIMD is only supported on arm64 when using the LLVM backend. When not using
+ * the LLVM backend, treat SIMD datatypes as regular value types.
+ */
+#define MONO_CLASS_IS_SIMD(cfg, klass) (((cfg)->opt & MONO_OPT_SIMD) && COMPILE_LLVM (cfg) && m_class_is_simd_type (klass))
 #else
-#define MONO_CLASS_IS_SIMD(cfg, klass) (((cfg)->opt & MONO_OPT_SIMD) && m_class_is_simd_type (klass))
+#define MONO_CLASS_IS_SIMD(cfg, klass) (((cfg)->opt & MONO_OPT_SIMD) && m_class_is_simd_type (klass) && (COMPILE_LLVM (cfg) || mono_type_size (m_class_get_byval_arg (klass), NULL) == 16))
 #endif
```
Comment (vargaz) [inline src/mono/mono/mini/mini.h:326]: This means that non-arm64 only supports vector128 in non-llvm mode right ?

Comment (imhameed) [reply src/mono/mono/mini/mini.h:326]: Yeah. Maybe the condition should be made more specific, and MONO_CLASS_IS_SIMD should expand to a constant "false" expression on everything other than x86, amd64, and arm64.

**Review by vargaz — APPROVED** [review]
